src/core/server/saved_objects/import/regenerate_ids.ts

```diff
@@ -42,3 +44,36 @@ export const regenerateIds = (objects: SavedObject[]) => {
   }, new Map<string, { id: string; omitOriginId?: boolean }>());
   return importIdMap;
 };
+
+export const regenerateIdsWithReference = async (props: {
+  savedObjects: SavedObject[];
+  savedObjectsClient: SavedObjectsClientContract;
+  workspaces?: string[];
+  objectLimit: number;
+}): Promise<Map<string, { id?: string; omitOriginId?: boolean }>> => {
+  const { savedObjects, savedObjectsClient, workspaces, objectLimit } = props;
+  if (!workspaces || !workspaces.length) {
+    return savedObjects.reduce((acc, object) => {
+      return acc.set(`${object.type}:${object.id}`, { id: object.id, omitOriginId: false });
+    }, new Map<string, { id: string; omitOriginId?: boolean }>());
+  }
+
+  const allSavedObjects = (await exportSavedObjects({
```

Comment:
Why do we need exportSavedObjects again here?

Reply:
You are right, there is no need to call `exportSavedObjects` here, thanks for catching.